[AGENT-4246] Update documentation for new operators (#29)
datarobot/airflow-provider-datarobot: README.md

By default, all components use `datarobot_default` connection ID.

## Creating DataRobot preconfigured Connections

You can create preconfigured Connections to store and manage credentials 
that can be used together with Airflow Operators to replicate connection 
on DataRobot side ([Data connections](https://docs.datarobot.com/en/docs/data/connect-data/stored-creds.html)).

> Currently, supported the next types of credentials:

Reviewer: ```suggestion
Currently supported types of credentials:
```

* `DataRobot Basic Credentials` - to store login/password pairs
* `DataRobot GCP Credentials` - to store Google Cloud Service account key
* `DataRobot AWS Credentials` - to store AWS access keys
* `DataRobot Azure Storage Credentials` - to store Azure Storage secret
* `DataRobot OAuth Credentials` - to store OAuth tokens
* `DataRobot JDBC DataSource` - to store JDBC connection attributes

> After creating preconfigured Connections using Airflow UI or Airflow API [Managing Connections](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html)

Reviewer: ```suggestion
After creating preconfigured connections using Airflow UI or Airflow API [Managing Connections](https://airflow.apache.org/docs/apache-airflow/stable/howto/connection.html)
```

> it can be used with GetOrCreateCredentialOperator or GetOrCreateDataStoreOperator

Reviewer: ```suggestion
it can be used with `GetOrCreateCredentialOperator` or `GetOrCreateDataStoreOperator`
```

> to replicate it on DataRobot side and retrieve corresponding credentials_id
> or datastore_id. 

Reviewer: ```suggestion
to replicate it on DataRobot side and retrieve corresponding `credentials_id`
or `datastore_id`. 
```

Can you also make a warning that credentials are only created one time -- if the value of the credential is changed in Airflow secret storage, they must be manually updated in DataRobot Credentials Manager to keep them in sync.

Author: I will add this in separate a commit

## Modules

### [Operators](https://github.com/datarobot/airflow-provider-datarobot/blob/main/datarobot_provider/operators/)

- `GetOrCreateCredentialOperator`

>     Fetching credentials by Credential name and return Credentials ID. 

Reviewer: ```suggestion
    Fetching credential by name and return `credential_id`. 
```

>     I case if credential does not exist it will try to create it using Airflow preconfigured connection
>     with the same connection name.
>  
>     Required config params:
> 
>         credentials_param_name: str - name of parameter in the config file to get Credential name
> 
>     Returns a credential ID.

Reviewer: ```suggestion
    Attempt to find DataRobot credential with the provided name and in case if credential does not exist it will create it using Airflow preconfigured connection
    with the same connection name.
 
    Required config params:

        `credentials_param_name`: str - name of parameter in the config file to get Credential name

    Returns a credential ID.
```

- `GetOrCreateDataStoreOperator`

    Fetching DataStore by Connection name and return DataStore ID. 

>     I case if DataStore does not exist it will try to create it using Airflow preconfigured connection
>     with the same connection name.
>  
>     Required config params:
> 
>         connection_param_name: str - name of parameter in the config file to get Connection name

Reviewer: ```suggestion
    In case if DataStore does not exist it will try to create it using Airflow preconfigured connection
    with the same connection name.
 
    Required config params:

        `connection_param_name`: str - name of parameter in the config file to get Connection name
```

    Returns a credential ID.

- `CreateDatasetFromDataStoreOperator`

    Loading dataset from JDBC Connection to DataRobot AI Catalog and return Dataset ID
 
    Required config params:

>         datarobot_jdbc_connection: str - existing of preconfigured Datarobot Connection name
>         dataset_name: str - name of loaded dataset,
>         table_schema: str - database table schema,
>         table_name: str - source table name,
>         do_snapshot: bool - If unset, uses the server default: True. If true, creates a snapshot dataset; if false, creates a remote dataset. Creating snapshots from non-file sources may be disabled by the permission, Disable AI Catalog Snapshots.
>         persist_data_after_ingestion: bool - If unset, uses the server default: True. If true, will enforce saving all data (for download and sampling) and will allow a user to view extended data profile (which includes data statistics like min/max/median/mean, histogram, etc.). If false, will not enforce saving data. The data schema (feature names and types) still will be available. Specifying this parameter to false and doSnapshot to true will result in an error.

Reviewer: ```suggestion
        - `datarobot_jdbc_connection`: str - existing of preconfigured Datarobot Connection name
        - `dataset_name`: str - name of loaded dataset
        - `table_schema`: str - database table schema
        - `table_name`: str - source table name
        - `do_snapshot`: bool - If unset, uses the server default(`True`). If `True`, creates a snapshot dataset; if `False`, creates a remote dataset. Creating snapshots from non-file sources may be disabled by the permission, _Disable AI Catalog Snapshots_.
        - `persist_data_after_ingestion`: bool - If unset, uses the server default(`True`). If `True`, will enforce saving all data (for download and sampling) and will allow a user to view extended data profile (which includes data statistics like min/max/median/mean, histogram, etc.). If `False`, will not enforce saving data. The data schema (feature names and types) still will be available. Specifying this parameter to **false** and `doSnapshot` to **true** will result in an error.
```

    Scores batch predictions against the deployment.

    Prerequisites:

>     - You can use GetOrCreateCredentialOperator to pass `credential_id` from preconfigured DataRobot Credentials (Airflow Connections)

Reviewer: ```suggestion
    - You can use `GetOrCreateCredentialOperator` to pass `credential_id` from preconfigured DataRobot Credentials (Airflow Connections)
```

      or you can manually set `credential_id` parameter in the config. [S3 credentials added to DataRobot via Python API client](https://datarobot-public-api-client.readthedocs-hosted.com/en/latest-release/reference/admin/credentials.html#s3-credentials).
    - OR a Dataset ID in the AI Catalog

>     - OR a DataStore ID for jdbc source connection, you can use GetOrCreateDataStoreOperator to pass `datastore_id` from preconfigured Airflow Connection

Reviewer: ```suggestion
    - OR a DataStore ID for jdbc source connection, you can use `GetOrCreateDataStoreOperator` to pass `datastore_id` from preconfigured Airflow Connection
```
